great_expectations/rule_based_profiler/config/base.py

```diff
@@ -167,6 +168,11 @@ def __init__(
         if class_name is not None:
             self.class_name = class_name
 
+        if evaluation_parameter_builder_configs:
+            self.evaluation_parameter_builder_configs = (
+                evaluation_parameter_builder_configs
+            )
+
```

Comment:
I really would prefer not to conditionally set attributes. ALL instances of a class should have the same shape, with individual instances having different values.

By conditionally setting this attr, we violate a core principle of class-based development.

Reply:
same with class_name and module_name above? I agree, and if done for serialization purposes we should use schema to suppress writing attributes with None values

Reply:
@cdkini The logic tries to say "only do it for metric builder".  Let me see what can be done here...

Reply:
@cdkini I think that this is totally fine here -- we do not want empty (`None`) elements having to watch for in all tests.  We use this pattern in other places.  If we did not, the very `great_expectations.yml` would be full of keys with no value.

Reply:
@cdkini @donaldheppner I am making an attempt to accomplish the same via `Schema`.

Reply:
@donaldheppner @cdkini  This approach is not yet possible, because we do not always use `Schema.dump()` for serializing (we sometimes use `to_json_dict()`, which can be custom).  This is because we do not yet have a tight Marshmallow `Schema` for all configuration objects — for example, `ExpectationConfigurationBuilder` (and, correspondingly, `ExpectationConfigurationBuilderConfig`) can allow any `kwargs`.  With Marshmallow `Schema.dump() `all of them would be lost.  Once we implement a better `Schema` (with inheritance), this will be possible.  Thanks.

Reply:
@donaldheppner @cdkini @anthonyburdi Seems like I was able to get it to work with having to fix only one test.  Thanks!  P.S.: We still have to address the broader deficiency in Marshmallow `Schema` usage, alluded to above.